cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/optimizer/DataTypeRequirementsProcessorTest.java

```diff
@@ -0,0 +1,216 @@
+                basePath + "/alltypes/modelinfo/alltypes-modelinfo-1.0.0.xml", null);
+        
+        Map<String,Set<StringMatcher>> expectations = new HashMap<>();
+        expectations.put("A", new HashSet<>(Arrays.asList(new PrefixStringMatcher("code_col"))));
+        
+        assertEquals( expectations, reqsByDataType );
+    }
+
+// This test case only matters if the user is allowed to pass non-null expression
+// lists and that is currently blocked.
+//    @Test
+//    public void testMissingExpression() throws Exception {
+//        IllegalArgumentException ex = assertThrows( IllegalArgumentException.class,
+//                () -> runTest("src/test/resources/fhir/cql", null, Collections.singleton("DoesNotExist")));
+//        
+//        assertEquals( "One or more requested expressions [DoesNotExist] not found in library {\"libraryId\":\"FhirTest\",\"version\":\"1.0.0\",\"format\":\"ELM\"}", ex.getMessage() );
+//    }
+    
+    protected Map<String, Set<String>> runPathTest(String cqlPath, String modelInfoPath, Set<String> expressions) throws IOException, FileNotFoundException {
+        return runPathTest(cqlPath, modelInfoPath, expressions, null);
+    }
+
+    protected Map<String, Set<String>> runPathTest(String cqlPath, String modelInfoPath, Set<String> expressions, Predicate<CqlLibraryDescriptor> libraryFilter) throws IOException, FileNotFoundException {
+        CqlLibraryProvider cpBasedLp = new ClasspathCqlLibraryProvider("org.hl7.fhir");
+        DirectoryBasedCqlLibraryProvider dirBasedLp = new DirectoryBasedCqlLibraryProvider(new File(cqlPath));
+        PriorityCqlLibraryProvider lsp = new PriorityCqlLibraryProvider(dirBasedLp, cpBasedLp);
+        
+        CqlToElmTranslator translator = new CqlToElmTranslator();
+        
+        if( modelInfoPath != null ) {
+            final File modelInfoFile = new File(modelInfoPath);
+            try( Reader r = new FileReader( modelInfoFile ) ) {
+                translator.registerModelInfo(r);
+            }
+        }
+        
+        DataTypeRequirementsProcessor requirementsProcessor = new DataTypeRequirementsProcessor(translator);
+        
+        Map<String,Set<String>> pathsByDataType = new HashMap<>();
+        for( CqlLibraryDescriptor cld : lsp.listLibraries() ) {
+            if( libraryFilter == null || libraryFilter.test(cld) ) {
+                System.out.println("Processing " + cld.toString());
+                Map<String,Set<String>> newPaths = requirementsProcessor.getPathsByDataType(lsp, cld);
+                
+                newPaths.forEach( (key,value) -> {
+                    pathsByDataType.merge(key, value, (prev,current) -> { prev.addAll(current); return prev; } );
+                });
+            } else {
+                System.out.println("Skipping " + cld.toString());
+            }
+        }   
+        
+        System.out.println(pathsByDataType);
```

Comment:
Pedantic: Were these println's leftover from debugging?
I'm not the biggest fan of our tests writing to stdout, but I'm OK with it if you think they should be there.

Reply:
I'm fine leaving it if you get value out of it.  Just wanted to be sure it wasn't leftover from the initial development of the tests.